src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -3250,6 +3250,99 @@ class AlterStageSegment(BaseSegment):
     )
 
 
+@snowflake_dialect.segment()
+class AppendOnlyExpressionSegment(BaseSegment):
+    """An APPEND_ONLY = TRUE | FALSE expression."""
+
+    type = "append_only_expression"
+    match_grammar = "APPEND_ONLY"
+
+    parse_grammar = Sequence(
+        "APPEND_ONLY",
+        Ref("EqualsSegment"),
+        OneOf(Ref("TrueSegment"), Ref("FalseSegment"))
+    )
+
+
+@snowflake_dialect.segment()
+class ShowInitialRowsExpressionSegment(BaseSegment):
+    """A SHOW_INITIAL_ROWS = TRUE | FALSE expression."""
+
+    type = "show_initial_rows_expression"
+    match_grammar = "SHOW_INITIAL_ROWS"
+
+    parse_grammar = Sequence(
+        "SHOW_INITIAL_ROWS",
+        Ref("EqualsSegment"),
+        OneOf(Ref("TrueSegment"), Ref("FalseSegment"))
+    )
+
+
+@snowflake_dialect.segment()
+class InsertOnlyExpressionSegment(BaseSegment):
+    """A INSERT_ONLY = TRUE expression."""
+
+    type = "insert_only_expression"
+    match_grammar = "INSERT_ONLY"
+
+    parse_grammar = Sequence(
+        "INSERT_ONLY",
+        Ref("EqualsSegment"),
+        Ref("TrueSegment")
+    )
+
+
+@snowflake_dialect.segment()
+class CreateStreamStatementSegment(BaseSegment):
+    """A Snowflake `CREATE STREAM` statement.
+
+    https://docs.snowflake.com/en/sql-reference/sql/create-stream.html
+    """
+
+    type = "create_stream_statement"
+
+    match_grammar = Sequence(
+        "CREATE", "STREAM"
+    )
+
+    parse_grammar = Sequence(
+        "CREATE",
+        Ref("OrReplaceGrammar", optional=True),
+        "STREAM",
+        Ref("IfNotExistsGrammar", optional=True),
+        Ref("ObjectReferenceSegment"),
+        Sequence("COPY", "GRANTS", optional=True),
+        "ON",
+        OneOf(
+            Sequence(
+                OneOf("TABLE", "VIEW"),
+                Ref("ObjectReferenceSegment"),        
+                OneOf(
+                    Ref("FromAtExpressionSegment", optional=True),
+                    Ref("FromBeforeExpressionSegment", optional=True)
+                ),
```

Comment:
Here's another thing you need to fix as well:

```suggestion
                OneOf(
                    Ref("FromAtExpressionSegment"),
                    Ref("FromBeforeExpressionSegment"),
                    optional=True
                ),
```

You'd think they would be equivalent, but they are not quite the same and with `OneOf` without the `optional=True` something has to match it (which it doesn't for your simple test case).

Reply:
Bingo. This was the culprit. I'm uncommenting the rest of my test fixtures and testing locally.